**DSCI_100 Individual Project Planning Stage**

In [6]:
library(tidyverse)
library(readxl)
library(readr)
library(tidymodels)
library(repr)

In [7]:
url <- "https://raw.githubusercontent.com/GSinghData/Convert-to-URL/refs/heads/main/players.csv"
raw_data <- read_csv(url)


Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [8]:
selected_data <- raw_data|>
    select(-hashedEmail)
selected_data

experience subscribe played_hours name      gender            Age
1   Pro         TRUE     30.3         Morgan    Male               9 
2   Veteran     TRUE      3.8         Christian Male              17 
3   Veteran    FALSE      0.0         Blake     Male              17 
4   Amateur     TRUE      0.7         Flora     Female            21 
5   Regular     TRUE      0.1         Kylie     Male              21 
6   Amateur     TRUE      0.0         Adrian    Female            17 
7   Regular     TRUE      0.0         Luna      Female            19 
8   Amateur    FALSE      0.0         Emerson   Male              21 
9   Amateur     TRUE      0.1         Natalie   Male              47 
10  Veteran     TRUE      0.0         Nyla      Female            22 
11  Veteran     TRUE      1.6         Lane      Female            23 
12  Pro         TRUE      0.0         Daniela   Male              17 
13  Beginner    TRUE      1.5         Sarah     Female            25 
14  Amateur     TRUE      0.2         Thatcher  Male              22 
15  Veteran     TRUE      0.0         Niamh     Non-binary        17 
16  Amateur    FALSE      0.0         Quinlan   Male              22 
17  Beginner    TRUE      0.0         Elodie    Female            17 
18  Amateur     TRUE     48.4         Xander    Female            17 
19  Amateur     TRUE      0.5         Marley    Male              17 
20  Regular     TRUE      0.6         Ryker     Male              19 
21  Regular     TRUE      0.3         Andy      Male              58 
22  Amateur    FALSE      0.1         Anastasia Female            17 
23  Beginner    TRUE      1.0         Leah      Male              17 
24  Beginner    TRUE      0.0         Ren       Male              17 
25  Amateur     TRUE      0.7         Hugo      Female            21 
26  Regular     TRUE      0.6         Kendall   Female            28 
27  Regular     TRUE      0.0         Iman      Male              17 
28  Veteran    FALSE      0.0         Finn      Male              23 
29  Amateur     TRUE      1.8         Luca      Male              23 
30  Veteran    FALSE      0.1         Vivienne  Male              18 
⋮   ⋮          ⋮         ⋮            ⋮         ⋮                 ⋮  
167 Amateur    FALSE      1.0         Radwan    Male              26 
168 Beginner   FALSE      0.3         Ariana    Female            17 
169 Veteran     TRUE      0.1         Quinton   Male              17 
170 Veteran    FALSE      0.1         Rocco     Male              17 
171 Pro        FALSE      0.4         Lyra      Male              21 
172 Beginner   FALSE      1.8         Amelia    Male              32 
173 Veteran     TRUE      0.0         Saif      Agender           20 
174 Regular    FALSE      0.0         Balthazar Male              50 
175 Veteran     TRUE      0.0         Parker    Male              17 
176 Amateur     TRUE      0.3         Joaquim   Male              17 
177 Pro        FALSE      0.0         Jesse     Non-binary        17 
178 Veteran    FALSE      2.7         Jamie     Non-binary        21 
179 Amateur     TRUE      0.4         Charlie   Two-Spirited      17 
180 Veteran     TRUE      0.1         Finnian   Non-binary        17 
181 Amateur    FALSE      2.1         Sebastián Male              24 
182 Amateur     TRUE      0.8         Hunter    Female            22 
183 Pro        FALSE      0.2         Liam      Male              17 
184 Amateur     TRUE     32.0         Sidney    Male              22 
185 Pro        FALSE      1.7         Asher     Male              17 
186 Regular    FALSE      0.1         Sam       Male              18 
187 Veteran     TRUE      0.1         Gabriela  Female            44 
188 Amateur     TRUE      0.0         Jasper    Male              17 
189 Beginner    TRUE      0.0         Lina      Female            17 
190 Amateur    FALSE      0.0         Orion     Male              17 
191 Amateur     TRUE      0.0         Rhys      Male              20 
192 Amateur   

**Data Description**
The above data has 196 unique observations. It contains 7 different variables named: experience (character), subscribe (logical), hashedEmail (character), played_hours (double), name (character), gender (character), and age (double). The experience variable is a description of how experienced each user is at the game of Minecraft. It is a list of an arbitrary size, with the possible values being: Beginner, Amateur, Regular, Veteran, and Pro. The subscribe variable, which is either true or false, records whether or not each user is subscribed to newletters from the team performing the study. The hashedEmail variable represents each user's email used to sign up without giving out their personal information. It is unique for each user and consists of letters and numbers. The played_hours variable keeps track of how many hours each user has played on the server for. The name variable is simply the name of the user. The gender variable is a catergory that is to be filled in by each user. Because each user identifies themselves in a way that they feel best represents them, and this is not bound to a limited number of possibilities, the number of possibilities for this variable are infinite, but the values that have been observed so far include: Male, Female, Non-Binary, Agender, Two-Spirited, Prefer not to say, and Other. The final variable, age, simply records the age of each user in years. The value of this column can be either a number, or NA.

A potential issut that could arrive from this data is the presence of the NA value in the age column. If someone were using the data from this column, they would have to either exclude the column by setting na.rm = TRUE, or calculate the value by averaging other ages that were close to that value on the plot.



**Question**
The Broad Question that I will be addressing is: We would like to know which "kinds" of players are most likely to contribute large amounts of data so we can target those players in our recruiting efforts.